### ISSUE TITLE
Observable differences in 2017 model B1248 (actually a GD1144 mislabelled)

### ISSUE THREAD

**Issue by Danimations** [Effix/LedBadge#10]
My B1248 unit arrived today, and I've noted one external difference from the unit you have pictured. 

My B1248* has a female microUSB (micro-B) USB 2.0 socket in place of the previous model's miniUSB (mini-B) USB 2.0 socket. Both are five pin connections, so hopefully the internal components don't deviate from your existing unit.

I'm yet to open the case to determine if there are any other differences among the internal components.

I can't tell you when this variation was manufactured, but I can tell you that the software was bundled in March 2017, according to the provided mini CD-ROM.

*_Note: on 9 June 2018, I realised that the internal PCB was actually an 1144 which had been incorrectly labelled as a 1248._

**Comment by Effix** [comment]
The usb socket isn't much to go on by itself. I've run into one that was wildly different on the inside and even looked nearly the same on the outside.

**Comment by Danimations** [comment]
Well, I got the badge open (though the case is a little damaged now and a button popped off in the process). 

The board looks much simpler than the earlier model. Labelling suggests it was assembled in 2017, consistent with the supplied software. 

There's only one part with a complex identifying code printed on it, which I took a close-up photo of. 

All the other components look simple by comparison (resistors, switches, capacitors etc.).

ARM chip appears to be labelled as follows:

GD32F150
C6T6
CE7F033
AJL721 ARM

Images of board and sole chip below:

![b1248 2017 model - led badge chip close-up](https://user-images.githubusercontent.com/39525759/41107247-2052ca3a-6ab1-11e8-9fe1-325228ee5ba9.JPG)
![b1248 2017 model - led badge internals](https://user-images.githubusercontent.com/39525759/41107250-20aa2988-6ab1-11e8-8cc5-70602a2898b8.jpg)

**Comment by Danimations** [comment]
I hope this manual is of some use to you: https://datasheet.lcsc.com/szlcsc/GD32F150C8T6_C80736.pdf

I don't know what to make of the remaining part #s listed below.

C6T6
CE7F033
AJL721 ARM

**Comment by Danimations** [comment]
Board is marked: TBD-1144-GDV1*

*_Note: after discovering that the LED matrix was actually 44 pixels wide and 11 pixels high, this PCB part # makes sense!_

**Comment by Effix** [comment]
Hmm... that's not great news. It isn't the same processor family, so the code won't be compatible.

It is a pretty beefy chip with even more ram and storage than the other one, but it is doing extra work, so it isn't clear off the bat if multiple brightness values are doable (Though, does it have a brightness button on it? That would be a hint.)

**Comment by Danimations** [comment]
Yes, it had a four-step brightness button... with 25%, 50%, 75% and 100%

**Comment by Danimations** [comment]
This is the eBay seller I bought mine from, in case you're interested in playing with this hardware yourself, @Effix .

https://www.ebay.com.au/itm/LED-Programmable-Scrolling-Name-Message-Badge-Tag-Digital-Display-English-5Color/263338381497?ssPageName=STRK%3AMEBIDX%3AIT&var=562350867206&_trksid=p2057872.m2749.l2649

**Comment by Effix** [comment]
That one is going to take some time investment to investigate, especially since there doesn't appear to be an easily accessible programming header (plus it will probably take some soldering and other tools that I don't currently have).

I'm definitely not going to be able to commit to it in the near term.

It's unfortunate that the picture of the board in the ebay listing looks like it would almost be compatible (it kind of looks like a mix of the two variants that are supported).

**Comment by Danimations** [comment]
Perhaps I can shop around in the hopes of finding a match or at least a closer match to the units you are working with? I'll open each one as it arrives and see what each contains.

**Comment by Danimations** [comment]
My next batch of badges arrived today. The three boxes were all marked 2017, but one of the badges had some noticeable external differences. On the yellow badge, both parts of the magnet clasp were mounted _outside_ the case (with the base plate glued on), and the pin was a separate part (white moulded plastic) compared with the pin clasps on the others which are nicely integrated into the mould of the back cover of the badge. So I might have one older model unit out of the four I've received thus far. I'm expecting three more, from three different sellers... so hopefully I will have a lucky strike with at least one of them.

I'll open the yellow one up later tonight and see if external moulding and clasps is where the differences end. Here's hoping I have something closer to yours on the inside, @Effix .

**Comment by Danimations** [comment]
I have some news! After closer inspection of the badges that have arrived thus far (all from one eBay seller)... I discovered that they actually sent me the wrong product, incorrectly labelled! They were boxed and branded as B1248's, but in actual fact they were 1144's... lower resolution, and completely different circuitry.

This actually makes me _more hopeful_ that a match or near match on a 1248 is possible with post-2016 badges. Now the part number printed on the PCB makes perfect sense!
